feat(anthropic-ai): implemented anthropic provider

VenomHare opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://voltagent.dev/docs/community/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [x] Tests for the changes have been added
- [x] Docs have been added / updated
- [x] Changesets have been added: https://voltagent.dev/docs/community/contributing/#creating-a-changeset

## What is the current behavior?
No Direct Implementation for Anthropic SDK

## What is the new behavior?
Allows users to use `AnthropicProvider` as LLM while creating an Agent

Fixes #10

omeraplak commented:
Hey @VenomHare ,
Thank you so much for the amazing contribution! 🙏
Would you mind pulling the latest changes from main and updating the `pnpm-lock.yaml` file as well?
That should help get the builds passing.

Really appreciate your help!

VenomHare commented:
Of course!
Let me know if there's anything else you'd like me to update.

omeraplak commented:
Thanks so much for the quick response!

It looks like the build is still failing: https://github.com/VoltAgent/voltagent/actions/runs/14714401255/job/41294436292?pr=58
Would you mind running pnpm install and updating the PR when you get a chance?

Really appreciate your help!

omeraplak commented on packages/anthropic-ai/src/index.spec.ts:
Can we use the "createTool" function here? https://voltagent.dev/docs/agents/tools/

VenomHare commented on packages/anthropic-ai/src/index.spec.ts:
Sure! 
I also used `createTool()` in example app

VenomHare commented:
I've added the createTool function for tool creation and fixed the issue where the ZodObject wasn't being recognized during parameter parsing. Take a look at the code now.

VenomHare commented:
Check now 
I have fixed the lock file

omeraplak commented:
Great, the build is working now! Congrats @VenomHare . I’m planning to merge it soon. I’m also doing a refactor on the core framework related to the types.

VenomHare commented:
Thank you so much for your cooperation.
As I recently started contributing to open source, I’ve learned a lot while working on this feature and had a great experience overall.

omeraplak commented:
Hey @VenomHare ,
We recently updated our provider docs (after you opened this PR).
Would you be open to updating your changes based on the new docs?

If you’re short on time, I can also handle it after merging, totally up to you!
https://voltagent.dev/docs/providers/overview/

VenomHare commented:
Done! Let me know if there's anything else you need. Happy to help!

omeraplak commented on website/docs/agents/providers.md:
```suggestion
```
Hey! I might be missing something, but I wasn’t totally sure why these lines were added, mind explaining a bit?

omeraplak commented:
Hey @VenomHare ,
Could you pull from main and run `pnpm install`? After that, we’re all set to merge! Thanks so much again 🙌

omeraplak commented on examples/with-anthropic/package.json:
```suggestion
    "zod": "3.24.2"
```

omeraplak commented on packages/anthropic-ai/package.json:
```suggestion
    "zod": "3.22.2"
```

omeraplak commented:
Hey @VenomHare,
The root cause was related to zod version 2.24.3 (issue: https://github.com/colinhacks/zod/issues/3435).
After that, a few more errors popped up. I’ve fixed those as well.
Really appreciate your awesome effort on this! 🙌 Merged 🎉
